hasher-matcher-actioner/hmalib/common/image_sources.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import functools
+import boto3
+import requests
+import typing as t
+from hmalib.common.message_models import (
+    S3ImageSubmission,
+    URLImageSubmissionMessage,
+)
+
+
+class ImageSource:
+    """
+    An interface that is used during hashing to get an image's bytes.
+    """
+
+    def get_image_bytes(self, identifier: str) -> bytes:
+        """
+        Get bytes for the image.
+        """
+        raise NotImplementedError
+
+
+@functools.lru_cache(maxsize=None)
+def _get_s3_client():
+    # memoized so that we don't create a boto3 client unless needed.
+    return boto3.client("s3")
+
+
+class S3BucketImageSource(ImageSource):
+    """
+    Get images from a single S3 bucket.
+
+    Potential enhancements:
+    - Customize retry behavior, with backoff
+    - Parameterized credentials rather than relying on boto inference
+    """
+
+    def __init__(self, bucket: str) -> None:
+        self.bucket = bucket
+
+    def get_image_bytes(self, identifier: str) -> bytes:
+        return (
+            _get_s3_client()
+            .get_object(Bucket=self.bucket, Key=identifier)["Body"]
+            .read()
+        )
+
+
+class URLImageSource(ImageSource):
+    """
+    Simple GET request to get bytes of a URL.
+
+    Potential enhancements:
+    - HTTP Retry configuration with backoff
+    - HTTP Authorization, Headers
+    - Customizable Keep-Alive handling (presently defaults to requests.Session
+      defaults)
+    """
+
+    def get_image_bytes(self, identifier: str) -> bytes:
+        r = requests.get(identifier)
+        r.raise_for_status()
+        return r.content
+
+
+def get_image_bytes(
```

Comment:
I don't see this method being used anywhere, was it for debug/future utility? 
can you delete or add a docstring? Since it shares the name of the class method a rename or suffix might be helpful.

Reply:
It is actually what the pdq_hasher lambda uses to obtain the image. Introduced in a later commit. I see your confusion. I'll add a docstring though.

https://github.com/facebook/ThreatExchange/pull/693/files#diff-dca029e7f27b7a31a5d85e9275abf7d49bc1535026d25e6173e4d4dcbe76cca3R78

Reply:
I missed this because I mistook `image_sources` of `image_sources.get_image_bytes(image)` as an object not a module. Given that we called the same methods on objects in the test. Can we please change the name of one or both?